dynatrace-aws-logs.sh

```diff
@@ -91,6 +91,19 @@ arguments:
     if [ -z $2 ] || [[ $2 == "--"* ]]; then echo "Missing value for parameter $1"; print_help_deploy; exit 1; fi
   }
 
+  function check_activegate_state() {
+    if ACTIVE_GATE_STATE=$(curl -ksS "${TARGET_URL}/rest/health" --connect-timeout 20); then
+      if [[ "$ACTIVE_GATE_STATE" != "RUNNING" ]]
+      then
+        echo -e ""
+        echo -e "\e[91mERROR: \e[37mActiveGate endpoint is not reporting RUNNING state. Please verify provided values for parameter: --target-url (${TARGET_URL})."
+        exit 1
+      fi
+    else
+        echo -e "\e[93mWARNING: \e[37mFailed to connect to ActiveGate url $TARGET_URL to check state. It can be ignored if ActiveGate does not allow public access."
```

Comment:
For AWS, existingAG option requires AG with public access. So this would be error + exit 1

Reply:
why then in line 120 we do not have exit?
should I change both places?

Reply:
Good question, I think in 120 the thought was that AG could be only exposing publicly one endpoint (for logs ingest). So following this approach this here should be a Warning.
To be honest I am not sure what approach we should take. It is tempting but also not most secure to ask for full public access. 
What do you think?

I think @mswiatkowska added somewhere another check that sends test log to given existingAG, is this check then not a duplicate?
Anyhow, assuming we have this check I described here it seems safe to leave this Warning as is.

Reply:
there are three check in case of using existing activegate to provide more detailed feedback to user what is not ok,
starting from basic check into final log ingest check (next check result depends on conditions checked in previous one):
check ActiveGte status (check_activegate_state)
check token permission (check_api_token)
try to ingest test log line (check_log_ingest_url)

Reply:
OK this looks very nice. We have error+exit 1 on check_log_ingest_url and it is 100% enough. So this line here is good as warning 👍